oss_src/sframe_query_engine/planning/optimizations/general_union_project_transforms.hpp

```diff
@@ -224,10 +224,16 @@ class opt_project_add_direct_source_tags : public opt_transform {
       auto nn = n->inputs[idx_1];
       
       if(nn->type == planner_node_type::SFRAME_SOURCE_NODE) {
-        input_mapping[i] = nn->any_p<sframe>("sframe").select_column(idx_2); 
+        auto sa = nn->any_p<sframe>("sframe").select_column(idx_2);
+        if (nn->p("begin_index") == 0 && nn->p("end_index") == sa->size()) {
```

Comment:
As far as I can tell input_mapping only goes into the field "direct_source_mapping" which is just column_id --> sarray.